zulip_bots/zulip_bots/bots/susi/susi.py

```diff
@@ -41,9 +41,53 @@ def handle_message(self, message: Dict[str, str], bot_handler: Any) -> None:
             return
         reply = requests.get("https://api.susi.ai/susi/chat.json", params=dict(q=msg))
         try:
-            answer = reply.json()['answers'][0]['actions'][0]['expression']
-        except Exception:
+            data = reply.json()
+            actions = data['answers'][0]['actions']
+            answer = ""
+            for action in actions:
+                if action['type'] == 'answer':
+                    answer+=action['expression']+'\n'
+                elif action['type'] == 'anchor':
+                    text=action['text'];
+                    link=action['link'];
+                    answer+='['+text+']('+link+')'+'\n';
+                elif action['type'] == 'table':
+                    tableData = data['answers'][0]['data']
+                    msg=""
+                    columns = list(action['columns'].keys())
+                    columnsData = list(action['columns'].values())
+                    # using markdown make a table
+                    for heading in columnsData:
+                        answer+="| "+heading+" "
+                    answer+="\n"
+                    for heading in columns:
+                        answer+="|"+"------"+" "
+                    answer+="\n"
+                    for item in (tableData):
+                        if(item[columns[0]]):
+                            column1=""
```

Comment:
ideally, >3 indentations - and I usually call it - `something smells not clean`. Probably break it down at some point - return early, invert if, etc ?

Reply:
@tonythomas01 refactored the code to remove the if statements. It it okay now?